From f3c65536e02764486f1dd0082c0537377ee8720b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 27 Mar 2014 15:28:25 +0000
Subject: [PATCH] mkinitfs: fix kernel module dependency tracing

When searching kernel module depedencies we need to that relative
kerneldir and not relative basedir.
---
 mkinitfs.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mkinitfs.in b/mkinitfs.in
index cd10359..fc7cf34 100755
--- a/mkinitfs.in
+++ b/mkinitfs.in
@@ -68,7 +68,7 @@ function recursedeps(k,		j, dep) {
 	split(deps[k], dep, " ");
 	for (j in dep)
 		recursedeps(dep[j]);
-	print(k);
+	print(prepend k);
 }
 
 BEGIN {
@@ -98,7 +98,7 @@ find_kmods() {
 	local oldpwd="$PWD"
 	cd "$kerneldir" || return 1
 	for file in $(feature_files "${kerneldir}/" modules); do
-		echo ${file#$basedir}
+		echo ${file#${kerneldir%/}/}
 	done | find_kmod_deps
 	cd "$oldpwd"
 }
-- 
1.9.1

